src/parser/syntax.rs

```diff
@@ -282,63 +384,73 @@ pub fn parse_list<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_, F, Syn
     }
 }
 
-pub fn parse_quote<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_, F, Syntax<F>> {
+pub fn parse_quote<F: LurkField>(
+    state: Rc<RefCell<State>>,
+    create_unknown_packages: bool,
+) -> impl Fn(Span<'_>) -> ParseResult<'_, F, Syntax<F>> {
     move |from: Span<'_>| {
         let (i, c) = opt(parse_char())(from)?;
         if let Some(c) = c {
             Ok((i, c))
         } else {
             let (i, _) = tag("'")(from)?;
-            let (upto, s) = parse_syntax()(i)?;
+            let (upto, s) = parse_syntax(state.clone(), false, create_unknown_packages)(i)?;
             let pos = Pos::from_upto(from, upto);
             Ok((upto, Syntax::Quote(pos, Box::new(s))))
         }
     }
 }
 
 // top-level syntax parser
-pub fn parse_syntax<F: LurkField>() -> impl Fn(Span<'_>) -> ParseResult<'_, F, Syntax<F>> {
+pub fn parse_syntax<F: LurkField>(
+    state: Rc<RefCell<State>>,
+    meta: bool,
+    // this parameter triggers a less strict mode for testing purposes
+    create_unknown_packages: bool,
+) -> impl Fn(Span<'_>) -> ParseResult<'_, F, Syntax<F>> {
     move |from: Span<'_>| {
         alt((
-            parse_hash_char(),
+            context(
+                "list",
+                parse_list(state.clone(), meta, create_unknown_packages),
+            ),
             parse_uint(),
             parse_num(),
-            context("symbol", parse_symbol()),
+            context(
+                "symbol",
+                parse_symbol(state.clone(), create_unknown_packages),
+            ),
             parse_string(),
-            context("quote", parse_quote()),
-            context("list", parse_list()),
+            context("quote", parse_quote(state.clone(), create_unknown_packages)),
+            parse_hash_char(),
         ))(from)
     }
 }
 
 pub fn parse_maybe_meta<F: LurkField>(
+    state: Rc<RefCell<State>>,
+    create_unknown_packages: bool,
 ) -> impl Fn(Span<'_>) -> ParseResult<'_, F, Option<(bool, Syntax<F>)>> {
     move |from: Span<'_>| {
         let (_, is_eof) = opt(nom::combinator::eof)(from)?;
         if is_eof.is_some() {
             return Ok((from, None));
         }
         let (next, meta) = opt(char('!'))(from)?;
-        if meta.is_some() {
-            let (end, syntax) = parse_syntax()(next)?;
-            Ok((end, Some((true, syntax))))
-        } else {
-            let (end, syntax) = parse_syntax()(from)?;
-            Ok((end, Some((false, syntax))))
-        }
+        let meta = meta.is_some();
+        let (end, syntax) = parse_syntax(state.clone(), meta, create_unknown_packages)(next)?;
+        Ok((end, Some((meta, syntax))))
     }
 }
 
 #[cfg(test)]
 pub mod tests {
```

Comment:
This updates a ton of hardcoded expected output. You could make sure you are the last one that has to suffer through this using [expect-test](https://docs.rs/expect-test/latest/expect_test/)

Reply:
If I'm gonna have to change everything to use that lib, then it's better to do it in another (focused) PR